lizaku/assignment3/assignment3.py

```diff
@@ -0,0 +1,127 @@
+class Dictionary:
+    '''
+    I implement the dictionary as a trie. As an input I give an array of words 
+    that are then transformed into a trie. isWord and isPrefix are methods of the Dictionary.
+    '''
+    def __init__(self, data):
+        self.data = self._make(data)
+    
+    def _make(self, words):
+        # constructing the trie, it's in form of a dictionary (not very efficient though...)
+        _end = '_end_'
+        root = dict()
+        for word in words:
+            current_dict = root
+            for letter in word:
+                current_dict = current_dict.setdefault(letter, {})
+            current_dict[_end] = _end
+        return root
+    
+    def isWord(self, word):
+        _end = '_end_'
```

Comment:
Just a note (no change required): here you use the string '_end_' as a marker. What you could do if you wanted to allow '_end_' as a value could be to have a specific Node object for each level of the trie and have the Node keep an attribute like is_word or is_final that is set to True only if you have reached the end of a full word.

Reply:
Yes, but I don't make a proper class for Trie here, though it would be a better decision.